lib/elastic_apm/config.rb

```diff
@@ -320,6 +331,24 @@ def build_logger
     def format_name(str)
       str.gsub('::', '_')
     end
+
+    def normalize_durations
+      DURATION_KEYS.each do |key|
+        value = send(key).to_s
+        default_unit = DURATION_DEFAULT_UNITS.fetch(key, 's')
+        duration = Duration.parse(value, default_unit: default_unit)
+        send("#{key}=", duration.minutes)
+      end
+    end
+
+    def normalize_sizes
+      SIZE_KEYS.each do |key|
+        value = send(key).to_s
+        default_unit = SIZE_DEFAULT_UNITS.fetch(key, 'b')
+        size = Size.parse(value, default_unit: default_unit)
+        send("#{key}=", size.bytes)
+      end
```

Comment:
You could add the default values directly to the hash definitions, and then only define one normalize function for everything, e.g.
```
DURATION_DEFAULT_UNITS = { span_frames_min_duration: 'ms' }
DURATION_DEFAULT_UNITS.defaults = 's'

def normalize_units(keys, default_units)
      keys.each do |key|
        value = send(key).to_s
        size = Size.parse(value, default_unit: default_units[key])
        send("#{key}=", size.val)
      end
```